extensions/notebook/src/jupyter/jupyterServerInstallation.ts

```diff
@@ -165,7 +175,28 @@ export class JupyterServerInstallation implements IJupyterServerInstallation {
 
 		try {
 			let pythonExists = await utils.exists(this._pythonExecutable);
-			if (!pythonExists || forceInstall) {
+			let upgradePython = false;
+			// Warn users that some packages may need to be reinstalled after updating Python versions
+			if (pythonExists && !this._usingExistingPython && await this.getInstalledPythonVersion(this._pythonExecutable) !== constants.pythonVersion) {
+				upgradePython = await vscode.window.showInformationMessage(msgPythonVersionUpdateWarning, yes, no) === yes;
+			}
+
+			if (!pythonExists || forceInstall || upgradePython) {
+				if (upgradePython) {
```

Comment:
This check doesn't seem correct - since it means that we'll essentially ignore the values of pythonExists/forceInstall

Reply:
This is for steps that are only applicable to updating Python (ex. prompting to remove old installation). The rest of the installation process starts at line 200.